lint/golangci-lint.bzl

```diff
@@ -0,0 +1,146 @@
+        arguments = [args],
+        mnemonic = _MNEMONIC,
+        tools = [executable],
+    )
+
+# buildifier: disable=function-docstring
+def _golangci_lint_aspect_impl(target, ctx):
+    if ctx.rule.kind not in ["go_binary", "go_library", "go_test"]:
+        return []
+
+    report, info = report_file(_MNEMONIC, target, ctx)
+    golangci_lint_action(ctx, ctx.executable._golangci_lint, ctx.rule.files.srcs, ctx.file._config_file, report, ctx.attr.fail_on_violation)
+    return [info]
+
+def golangci_lint_aspect(binary, config):
+    """A factory function to create a linter aspect.
+
+    Attrs:
+        binary: a golangci-lint executable.
+        config: the .golangci.yaml file
+    """
+    return aspect(
+        implementation = _golangci_lint_aspect_impl,
+        attrs = {
+            "fail_on_violation": attr.bool(),
+            "_golangci_lint": attr.label(
+                default = binary,
+                executable = True,
+                cfg = "exec",
+            ),
+            "_config_file": attr.label(
+                default = config,
+                allow_single_file = True,
+            ),
+        },
+        toolchains = ["@io_bazel_rules_go//go:toolchain"],
+    )
+
+def golangci_lint_binary(name):
+    """Wrapper around native_binary to select the correct golangci-lint executable for the execution platform."""
+    native_binary(
+        name = name,
+        src = select(
+            {
+                "@bazel_tools//src/conditions:linux_x86_64": "@golangci_lint_linux_x86_64//:golangci-lint",
+                "@bazel_tools//src/conditions:linux_aarch64": "@golangci_lint_linux_aarch64//:golangci-lint",
+                "@bazel_tools//src/conditions:darwin_x86_64": "@golangci_lint_macos_x86_64//:golangci-lint",
+                "@bazel_tools//src/conditions:darwin_arm64": "@golangci_lint_macos_aarch64//:golangci-lint",
+            },
+            no_match_error = "golangci-lint hasn't been fetched for your platform",
+        ),
+        out = "golangci-lint",
+        visibility = ["//visibility:public"],
+    )
+
+# buildifier: disable=function-docstring
+def fetch_golangci_lint():
+    http_archive(
+        name = "golangci_lint_linux_x86_64",
+        build_file_content = "exports_files([\"golangci-lint\"])",
+        strip_prefix = "golangci-lint-1.55.2-linux-amd64",
+        sha256 = "ca21c961a33be3bc15e4292dc40c98c8dcc5463a7b6768a3afc123761630c09c",
+        urls = ["https://github.com/golangci/golangci-lint/releases/download/v1.55.2/golangci-lint-1.55.2-linux-amd64.tar.gz"],
+    )
+    http_archive(
+        name = "golangci_lint_linux_aarch64",
+        build_file_content = "exports_files([\"golangci-lint\"])",
+        strip_prefix = "golangci-lint-1.55.2-linux-arm64",
+        sha256 = "8eb0cee9b1dbf0eaa49871798c7f8a5b35f2960c52d776a5f31eb7d886b92746",
+        urls = ["https://github.com/golangci/golangci-lint/releases/download/v1.55.2/golangci-lint-1.55.2-linux-arm64.tar.gz"],
```

Comment:
how often does it do releases? I suspect we'll need an update automation like we have for `ruff` but it can be a later PR.

Reply:
every 1-3 months. Is renovate configured in the repo? If so, it should handle the http_archives in a batch.

Reply:
Hm, yeah renovate is one possibility. In the ruff case you can see we mirror a list of versions so that users can pin. I can pick this up either way, thanks.